src/libstore/globals.cc

```diff
@@ -249,6 +249,17 @@ unsigned int MaxBuildJobsSetting::parse(const std::string & str) const
     }
 }
 
+unsigned int MaxSubstitutionJobsSetting::parse(const std::string & str) const
+{
+    if (str == "auto") return std::max(1U, std::thread::hardware_concurrency());
```

Comment:
Probably the default should not be `std::thread::hardware_concurrency()` because substitutions are network rather than CPU bound.

Reply:
Considering other comments, I removed the `MaxSubstitutionJobsSetting` and used `Setting<unsigned int>` instead. This means there's no more `auto` logic.